### ISSUE TITLE
REAC ID 2 for 2 simple chem reactions in .out

### ISSUE THREAD

**Issue by rmcdermo** [firemodels/fds#11847]
The REAC ID for the second reaction with `N_SIMPLE_CHEMISTRY_REACTIONS=2` does not seem to be initialized, I get garbage in the .out file.

```
&HEAD CHID='test' /
&MESH IJK=10,10,10, XB=-1,1,-1,1,0,2/
&TIME T_END=.1 /
&REAC N_SIMPLE_CHEMISTRY_REACTIONS=2, FUEL='METHANE'/ 
&TAIL/
```

**Comment by drjfloyd** [comment]
I'll take a look.  I made some recent edits to how reaction info was being written to .out. Firebot passed so I don't think I broke 2-step altogether.

**Comment by drjfloyd** [comment]
I just updated source, compiled debug, and ran and see this:

 Gas Phase Reaction Information

   Reaction 1

      Priority:                                                               1

      Fuel                                           Heat of Combustion (kJ/kg)
      METHANE                                                        30091.5278
      2-step reaction,  Total Heat of Combustion                     50027.3649

      Primitive Species Stoich. Coeff.
      Species ID                                                          Molar
      NITROGEN                                                         0.000000
      OXYGEN                                                          -1.333333
      WATER VAPOR                                                      2.000000
      CARBON MONOXIDE                                                  0.666667
      SOOT                                                             0.333333
      METHANE                                                         -1.000000

      Tracked (Lumped) Species Stoich. Coeff.
      Species ID                                             Molar         Mass
      AIR                                                -6.423882   -11.513156
      METHANE                                            -1.000000    -1.000000
      INTERMEDIATE PRODUCTS                               8.090548    12.513156

      Reaction Kinetics

      Fast chemistry

      ODE Solver:  EXPLICIT EULER

      Extinction Model:  EXTINCTION 1
      Auto-Ignition Temperature (C):            -273.1
      Critical Flame Temperature (C):           1507.0

      Prescribed Radiative Fraction:             0.200

   Reaction 2

      Priority:                                                               2

      Fuel                                           Heat of Combustion (kJ/kg)
      INTERMEDIATE PRODUCTS                                           1593.1902

      Primitive Species Stoich. Coeff.
      Species ID                                                          Molar
      OXYGEN                                                          -0.082401
      CARBON DIOXIDE                                                   0.123601
      CARBON MONOXIDE                                                 -0.082401
      SOOT                                                            -0.041200

      Tracked (Lumped) Species Stoich. Coeff.
      Species ID                                             Molar         Mass
      AIR                                                -0.396999    -0.460042
      INTERMEDIATE PRODUCTS                              -1.000000    -1.000000
      PRODUCTS                                            1.314598     1.460042

      Reaction Kinetics

      Fast chemistry

      ODE Solver:  EXPLICIT EULER

      Prescribed Radiative Fraction:             0.200

**Comment by rmcdermo** [comment]
For both "dv" and release I get something like this:

![Screenshot 2023-05-23 at 3 48 29 PM](https://github.com/firemodels/fds/assets/4418497/ed2270a1-4d0b-48b1-8522-510267ab1b56)

**Comment by drjfloyd** [comment]
Same build for me. Both dv and db seem to work.  Did you do a clean build?

**Comment by rmcdermo** [comment]
Marcos is getting the same as me on burn.  I'll try macOS

**Comment by mcgratta** [comment]
We should set ID='null' in the REACTION_TYPE definition. I'll do that

**Comment by drjfloyd** [comment]
OK. I see it now. Wasnt showing up in windows and wasn't showing when I mored the file to a putty terminal but it shows up if I open with Scite. Sounds like Kevin has it under control so I won't do anything yet.

**Comment by rmcdermo** [comment]
Well, does not seem like the last commit fixed it.  I'm still getting the problem in _db and _dv on burn.

With macOS things seem to work fine, and did before changing the type init.

**Comment by mcgratta** [comment]
I'll dig in more. Somehow that ID is begin overwritten with garbage

**Comment by mcgratta** [comment]
I do not understand the lines below:
```
         ! Set up paired reaction INTERMEDIATE PRODUCTS + AIR -> PRODUCTS
         ALLOCATE(REAC_TEMP(N_REACTIONS+NEW_REAC-1),STAT=IZERO)
         REAC_TEMP(1:NR) = REACTION(1:NR)
         IF (NEW_REAC > 1) REAC_TEMP(N_REACTIONS+1:N_REACTIONS+NEW_REAC-1) = REACTION(N_REACTIONS+1:N_REACTIONS+NEW_REAC-1)
         DEALLOCATE(REACTION)
         ALLOCATE(REACTION(N_REACTIONS+NEW_REAC),STAT=IZERO)
         REACTION(1:NR) = REAC_TEMP(1:NR)
         IF (NEW_REAC > 1) REACTION(N_REACTIONS+1:N_REACTIONS+NEW_REAC-1) = REAC_TEMP(N_REACTIONS+1:N_REACTIONS+NEW_REAC-1)
         DEALLOCATE(REAC_TEMP)
         RN2 => REACTION(REACTION(NR)%PAIR_INDEX)     <======== This line
         RN2 = REACTION(NR)                      <======== This line
         RN2%N_SIMPLE_CHEMISTRY_REACTIONS = -2
```

I think this is related to the problem

**Comment by drjfloyd** [comment]
When we have two-step chemistry we need to define a second reaction. Since we now allow for multiple simple chemistry reactions, we can't simply make the second step reaction index one more than the first step reaction index.  As we are processing the user specified &REAC, if there is a simple-chemistry with two-step we define the PAIR_INDEX. This is a pointer to the second step.  Since we don't read in the second step, we need to somehow intialize all the reaction variables with data that corresponds to the first step.  

RN2 => REACTION(REACTION(NR)%PAIR_INDEX)

In this line we have just processed REACTION(NR) where NR is a reaction that we have read in. It is a simple chemistry reaction with a second step. Now we need to define the second step. This second step (RN2) is REACTION(REACTION(NR)%PAIR_INDEX).

RN2 = REACTION(NR)  

Now that we have a pointer RN2 that is pointing to the second step we need to initialize it. The first part of that is to simply copy over everything that is in the first step reaction.  That is all this line is doing. 

The is here:

         IF (RN%ID=='null') THEN
            RN2%ID='null'
         ELSE
            RN2%ID=TRIM(RN%ID)//'_2'
         ENDIF


We no longer have RN pointing to a reaction since we have just reallocated the REACTION array to add an extra reaction for the second step. The two RN%ID should be REACTION(NR)%ID.
